SPECS/openmpi/openmpi.spec

```diff
@@ -0,0 +1,1139 @@
+BuildRequires:  libtool
+BuildRequires:  perl(Data::Dumper)
+BuildRequires:  perl(File::Find)
+%endif
+%if %{with rdma}
+BuildRequires:  opensm-devel
+BuildRequires:  rdma-core-devel
+%endif
+# So configure can find lstopo
+# BuildRequires:  hwloc-gui
+%if %{with java}
+BuildRequires:  java-devel
+%else
+Obsoletes:      %{name}-java < %{version}-%{release}
+Obsoletes:      %{name}-java-devel < %{version}-%{release}
+%endif
+%ifarch x86_64
+BuildRequires:  infinipath-psm-devel
+BuildRequires:  libpsm2-devel
+%endif
+%if %{with ucx}
+BuildRequires:  ucx-devel
+%endif
+
+%description
+Open MPI is an open source, freely available implementation of both the
+MPI-1 and MPI-2 standards, combining technologies and resources from
+several other projects (FT-MPI, LA-MPI, LAM/MPI, and PACX-MPI) in
+order to build the best MPI library available.  A completely new MPI-2
+compliant implementation, Open MPI offers advantages for system and
+software vendors, application developers, and computer science
+researchers. For more information, see http://www.open-mpi.org/ .
+
+%package devel
+Summary:        Development files for openmpi
+Requires:       %{name} = %{version}-%{release}
+Requires:       gcc-gfortran
+# Make sure this package is rebuilt with correct Python version when updating
+# Otherwise mpi.req from rpm-mpi-hooks doesn't work
+# https://bugzilla.redhat.com/show_bug.cgi?id=1705296
+Requires:       python(abi)
+Requires:       rpm-mpi-hooks
+Provides:       mpi-devel
+
+%description devel
+Contains development headers and libraries for openmpi.
+
+%if %{with java}
+%package java
+Summary:        Java library
+Requires:       %{name} = %{version}-%{release}
+Requires:       java-headless
+
+%description java
+Java library.
+
+%package java-devel
+Summary:        Java development files for openmpi
+Requires:       %{name}-java = %{version}-%{release}
+Requires:       java-devel
+
+%description java-devel
+Contains development wrapper for compiling Java with openmpi.
+%endif
+
+# We set this to for convenience, since this is the unique dir we use for this
+# particular package, version, compiler
+%global namearch openmpi-%{_arch}%{?_cc_name_suffix}
+
+%package -n python%{python3_pkgversion}-openmpi
+Summary:        OpenMPI support for Python 3
+Requires:       %{name} = %{version}-%{release}
+Requires:       python(abi)
+
+%description -n python%{python3_pkgversion}-openmpi
+OpenMPI support for Python 3.
+
+%prep
+%autosetup -p1 -n %{name}-%{version}
+%if %{with autogen}
+./autogen.pl --force
+%endif
+
+
+%build
+%{set_build_flags}
+./configure --prefix=%{_libdir}/%{name} \
+	--mandir=%{_mandir}/%{namearch} \
+	--includedir=%{_includedir}/%{namearch} \
+	--sysconfdir=%{_sysconfdir}/%{namearch} \
+	--disable-silent-rules \
+	--enable-builtin-atomics \
+	--enable-mpi-cxx \
+	--enable-ipv6 \
+%if %{with java}
+	--enable-mpi-java \
+	--with-jdk-bindir=%{_libdir}/jvm/msopenjdk-17/bin \
+	--with-jdk-headers=%{_libdir}/jvm/msopenjdk-17/include \
```

Comment:
I don't see any `msopenjdk-17` in our code, so I'd be careful here. It looks like `msopenjdk-11` should be safe.

Reply:
updated. 
I might have mixed msopenjdk and openjdk versions here.